Add 2545 natives.

4mmonium opened the pull request:
This should import all 2545 natives, I made a tool to import them, I compared the markdown, I hope I didn't miss anything 😀

blattersturm commented:
Why'd build fail here? Also, b2612 🙃

4mmonium commented:
> Is there any reason/purpose for `JOAAT_UNKNOWN`? Is the convention changing?

No, I thought it was a good idea since it makes it easier to find natives with missing joaat hashes (by searching JOAAT_UNKNOWN) later on and update them if needed.

4mmonium commented:
> > Is there any reason/purpose for `JOAAT_UNKNOWN`? Is the convention changing?
> 
> No, I thought it was a good idea since it makes it easier to find natives with missing joaat hashes (by searching `JOAAT_UNKNOWN`) later on and update them if needed. Done with the best intentions of course
> 
> For stuff like this for example: [d87e718](https://github.com/citizenfx/natives/commit/d87e718cc4a679ab5f175512a1cd7cc70e05c33b)
> 
> `JOAAT_UNKNOWN` only gets set when the `jhash` JSON field is empty. https://raw.githubusercontent.com/alloc8or/gta5-nativedb-data/master/natives.json

I corrected this, so instead of putting such in the native comment (potentially confusing users) I just placed a description for the native, telling the user the JOAAT hash is missing in case they want to fill it in.

**Old commit:** 
https://github.com/citizenfx/natives/commit/268b04141683d7d0f72cfaa4b8406ca788e5cb0f#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR8

**New commit:**
https://github.com/citizenfx/natives/pull/669/commits/8fca7b1a53ad4908ba2d2c2ecf9114fbc7a54579#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR12

AvarianKnight commented:
It would be useless for most end users anyways, plus with how native docs work now this would probably also give them `Has "modern" docs` even though they dont

4mmonium commented:
> > > > Is there any reason/purpose for `JOAAT_UNKNOWN`? Is the convention changing?
> > > 
> > > 
> > > No, I thought it was a good idea since it makes it easier to find natives with missing joaat hashes (by searching `JOAAT_UNKNOWN`) later on and update them if needed. Done with the best intentions of course
> > > For stuff like this for example: [d87e718](https://github.com/citizenfx/natives/commit/d87e718cc4a679ab5f175512a1cd7cc70e05c33b)
> > > `JOAAT_UNKNOWN` only gets set when the `jhash` JSON field is empty. https://raw.githubusercontent.com/alloc8or/gta5-nativedb-data/master/natives.json
> > 
> > 
> > I corrected this, so instead of putting such in the native comment (potentially confusing users) I just placed a description for the native, telling the user the JOAAT hash is missing in case they want to fill it in.
> > **Old commit:** [268b041#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR8](https://github.com/citizenfx/natives/commit/268b04141683d7d0f72cfaa4b8406ca788e5cb0f#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR8)
> > **New commit:** [8fca7b1#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR12](https://github.com/citizenfx/natives/commit/8fca7b1a53ad4908ba2d2c2ecf9114fbc7a54579#diff-26795edb097dd19bf2a1bcc4f6e49e50a246c8e763f0378b8c698153055651caR12)
> 
> this is kinda pointless, since for "brand-new" natives there will probably be no chance to get your hands on the jooat hash ever, recently chipjpg found some missing ones which were shipped in one of the last pushed xbox 360 builds (Title Update 27, 2016/2017)

Yeah, I did this taking into consideration what chipjpg did, but I removed it now since it doesn't make any sense at all and as @AvarianKnight said, it will just cause doc natives to show `Has "modern" docs` even though they don't.

gottfriedleibniz commented:
Is this ever going to be merged? 2545 was released ~18 months ago. Same w/ 2612 and 2802.

PsychoShock commented on VEHICLE/SetHeliResistToExplosion.md:
```diff
@@ -0,0 +1,18 @@
+
+## Parameters
+* **vehicle**:
+* **p1**:
```
P1 is if vehicle resist to explosion or not.

PsychoShock commented on TASK/TaskWarpPedDirectlyIntoCover.md:
````diff
@@ -0,0 +1,22 @@
+```
+
+## Parameters
+* **p0**:
+* **p1**:
+* **p2**:
+* **p3**:
+* **p4**:
+* **p5**:
````
* **ped**: The ped index.
* **canpeek**: If ped can peek and aim. False by default.
* **forceinitialdirection**: False by default.
* **forcefaceleft**: False by default. 
* **coverpoint**: Cover index.


I don't seem any `p5`.

PsychoShock commented on STATS/PlaystatsInstMissionEnd.md:
````diff
@@ -0,0 +1,17 @@
+// 0xFEA3F7E83C0610FA 
+void _PLAYSTATS_INST_MISSION_END(Any p0);
+```
+
````
This native activatesthe  following completion of instanced missions, commencing with the Fixer pack. This functionality isn't intended for competitive modes tho.

PsychoShock commented on STATS/PlaystatsHubExit.md:
````diff
@@ -0,0 +1,17 @@
+
+```c
+// 0x5A46ACE5C4661132 
+void _PLAYSTATS_HUB_EXIT(Any p0);
````
This is a list of the struct hub exit from what I found.

- `hubID`: Identifies this specific instance of the social hub, shared among players within it.
- `type`: Indicates the type of social hub the player departed from (e.g., music studio, smoking room).
- `bossId1` and `bossId2`: Unique identifiers for Boss instances present.
- `playerRole`: Denotes the player's role within the hub (e.g., CEO, bodyguard, biker prospect).
- `pulled`: Indicates if the player was forcibly removed from the social hub (e.g., due to a mission starting).
- `crewId`: Identifies the Social Club Crew to which the player belongs.
- `duration`: Records the time spent within the social hub.
- `dre`: Signifies the presence of Dr. Dre within the social hub.
- `playerCount`: Reflects the number of players present within the social hub at the time of exiting.

PsychoShock commented on OBJECT/N_0x5CE2E45A5CE2E45A.md:
````diff
@@ -0,0 +1,17 @@
+
+```c
+// 0x5CE2E45A5CE2E45A 
+void _0x5CE2E45A5CE2E45A(Any p0);
````
Name seems to be `FORCE_PORTABLE_PICKUP_LAST_ACCESSIBLE_POSITION_SETTING` with 1 param (The pickup id).

PsychoShock commented on NETWORK/N_0xFEA7A352DDB34D52.md:
````diff
@@ -0,0 +1,17 @@
+
+```c
+// 0xFEA7A352DDB34D52 
+void _0xFEA7A352DDB34D52(Any p0);
````
Seems to be named `NETWORK_BLOCK_PROXY_MIGRATION_BETWEEN_TUTORIAL_SESSIONS` with param BOOL if block or not. 

This native seems to prevents the transfer of network objects between players situated in different tutorial instances. I guess the flag needs to be deactivated when its purpose is fulfilled or no longer required.

PsychoShock commented on NETWORK/N_0xC6FCEE21C6FCEE21.md:
````diff
@@ -0,0 +1,22 @@
+
+```c
+// 0xC6FCEE21C6FCEE21 
+void _0xC6FCEE21C6FCEE21(Any p0);
````
Seems to be named `NETWORK_SET_TASK_CUTSCENE_INSCOPE_MULTIPLER` with float param for multiplier value. I bet it alters the scoping distance of `CTaskCutscene` for remote players. R* seems to reset to 1.0 when no longer needed.

PsychoShock commented on MONEY/NetworkSpentUpgradeAgency.md:
````diff
@@ -0,0 +1,20 @@
+```
+
+## Parameters
+* **p0**:
+* **p1**:
+* **p2**:
+* **p3**:
````
Param's seems to be BOOL `amount`, BOOL `if from bank AND wallet`, CHAR `context`, and CHAR `option` that seems to be 0 by default.

PsychoShock commented on MONEY/NetworkSpentSuvFastTravel.md:
````diff
@@ -0,0 +1,20 @@
+```
+
+## Parameters
+* **p0**:
+* **p1**:
+* **p2**:
+* **p3**:
````
Param's seems to be BOOL `amount`, BOOL `if from bank AND wallet` and CHAR `npcprovider`,

PsychoShock commented on MONEY/NetworkSpentSupplyContactService.md:
````diff
@@ -0,0 +1,20 @@
+```
+
+## Parameters
+* **p0**:
+* **p1**:
+* **p2**:
+* **p3**:
````
Param's seems to be BOOL `amount`, BOOL `if from bank`, BOOL `if from bank AND wallet` and `npcprovider`.

PsychoShock commented on HUD/SetCurrentCharacterHudColor.md:
````diff
@@ -0,0 +1,17 @@
+// 0x2ACCB195F3CCD9DE 
+void _SET_CURRENT_CHARACTER_HUD_COLOR(int hudColorId);
+```
+
````
This native enables the script to designate a personalized color for the multiplayer HUD, akin to modifying colors for characters in multiplayer. Need to clear this by passing -1 when no longer in use.

PsychoShock commented on GRAPHICS/N_0xC6730E0D14E50703.md:
````diff
@@ -0,0 +1,17 @@
+
+```c
+// 0xC6730E0D14E50703 
+void _0xC6730E0D14E50703(Any p0);
````
Correct name would be `SET_PTFX_FORCE_VEHICLE_INTERIOR_FLAG` with BOOL `value` to instructs all effect instances to set a flag, considering them as part of the vehicle interior.

PsychoShock reviewed (changes requested):
Old but can be good to know if it get merged one day!
